exp/controllers/azuremachinepool_controller.go

```diff
@@ -210,25 +211,57 @@ func (ampr *AzureMachinePoolReconciler) Reconcile(ctx context.Context, req ctrl.
 
 	logger = logger.WithValues("cluster", cluster.Name)
 
-	logger = logger.WithValues("AzureCluster", cluster.Spec.InfrastructureRef.Name)
-	azureClusterName := client.ObjectKey{
-		Namespace: azMachinePool.Namespace,
-		Name:      cluster.Spec.InfrastructureRef.Name,
-	}
-	azureCluster := &infrav1.AzureCluster{}
-	if err := ampr.Client.Get(ctx, azureClusterName, azureCluster); err != nil {
-		logger.V(2).Info("AzureCluster is not available yet")
-		return reconcile.Result{}, nil
-	}
+	var clusterScope azure.ClusterScoper
+	var groupScope groups.GroupScope
+	switch cluster.Spec.InfrastructureRef.Kind {
+	case "AzureCluster":
+		logger = logger.WithValues("AzureCluster", cluster.Spec.InfrastructureRef.Name)
+		azureClusterName := client.ObjectKey{
+			Namespace: azMachinePool.Namespace,
+			Name:      cluster.Spec.InfrastructureRef.Name,
+		}
+		azureCluster := &infrav1.AzureCluster{}
+		if err := ampr.Client.Get(ctx, azureClusterName, azureCluster); err != nil {
+			logger.V(2).Info("AzureCluster is not available yet")
+			return reconcile.Result{}, nil
+		}
 
-	// Create the cluster scope
-	clusterScope, err := scope.NewClusterScope(ctx, scope.ClusterScopeParams{
-		Client:       ampr.Client,
-		Cluster:      cluster,
-		AzureCluster: azureCluster,
-	})
-	if err != nil {
-		return reconcile.Result{}, err
+		// Create the cluster scope
+		scoper, err := scope.NewClusterScope(ctx, scope.ClusterScopeParams{
+			Client:       ampr.Client,
+			Cluster:      cluster,
+			AzureCluster: azureCluster,
+		})
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		clusterScope = scoper
+		groupScope = scoper
+	case "AzureManagedCluster":
+		logger = logger.WithValues("AzureManagedCluster", cluster.Spec.InfrastructureRef.Name)
+		azureManagedControlPlaneName := client.ObjectKey{
+			Namespace: azMachinePool.Namespace,
+			Name:      cluster.Spec.ControlPlaneRef.Name,
+		}
+		azureManagedControlPlane := &infrav1.AzureManagedControlPlane{}
+		if err := ampr.Client.Get(ctx, azureManagedControlPlaneName, azureManagedControlPlane); err != nil {
+			logger.V(2).Info("AzureManagedControlPlane is not available yet")
+			return reconcile.Result{}, nil
```

Comment:
Do we need to update the watches in `SetupWithManager` to include AzureManagedCluster/ControlPlane so we know this will eventually get requeued once the control plane becomes available?

Reply:
Added a watch on `AzureManagedCluster`